homeassistant/components/template/binary_sensor.py

```diff
@@ -388,6 +390,20 @@ def _handle_coordinator_update(self) -> None:
         """Handle update of the data."""
         self._process_data()
 
+        raw = self._rendered.get(CONF_STATE)
+        state = template.result_as_boolean(raw)
+
+        key = CONF_DELAY_ON if state else CONF_DELAY_OFF
+        delay = self._rendered.get(key) or self._config.get(key)
+
+        if (
+            self._delay_cancel
+            and delay
+            and self._attr_is_on == self._last_delay_from
+            and state == self._last_delay_to
+        ):
+            return
```

Comment:
I wonder if we should also check for `self.available`, in case that one changes to `False`?

Reply:
Although, now that I look at this, it's odd that binary_sensor overrides coordinator update

Reply:
I'm always willing to revisit sensor and binary_sensor.  Both of them do things differently than all other platforms.  Might be time to 'unify' all the systems for each platform.